.github/workflows/ci.yml

```
@@ -91,6 +91,27 @@ jobs:
     - name: Test on 🐍 3.10
       run: nox -s tests-3.10
 
+  cygwin:
+    name: Tests on 🐍 3.9 • cygwin
+    runs-on: windows-latest
+    needs: [lint]
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - uses: egor-tensin/setup-cygwin@v3
+      with:
+        platform: x64
+        packages: cmake gcc-g++ python39 python39-pip
+
+    - name: Install nox
+      run: python3.9 -m pip install nox
+
+    - name: Run tests
+      run: python3.9 -m nox -s tests-3.9
```

```suggestion
    - name: Run tests
      env:
        TMP: /tmp
        TEMP: /tmp
      run: python3.9 -m nox -s tests-3.9
```
A lot of the "This path doesn't match that path" problems look like they're in the platform temporary directory.  Changing that to a Cygwin directory might help with that.  Specifying the `shell` might also help.